blog/2020-07-13-alpine-python.md

```diff
@@ -0,0 +1,156 @@
+---
+author: Pascal Vizeli
+authorURL: https://github.com/pvizeli
+authorTwitter: pvizeli
+title: "Alpine with musl and Python"
+---
+
+It is well known that we use Alpine for most of our Containers. It is the perfect distribution for containers because it is small (BusyBox based), available for a lot of CPU architectures, and the package system is slim. Alpine and musl are new and had at least a significant development progress. That means many statements are today just wrong. I will clean up this incorrect information based on my research to provide the fastest way to run Home Assistant. This blogpost is not a musl vs. glibc flame ware; we use glibc on our OS. I will clean up old or half-truth, which is too long now in people's minds.
```

Comment:
Is it well known or should this be

We use Alpine for most of our containers.

Reply:
> Alpine and musl are new and had at least a significant development progress. That means many statements are today just wrong. I will clean up this incorrect information based on my research to provide the fastest way to run Home Assistant.

Alpine and musl are relatively new and have made a lot of progress since their starts. Because of this, many benchmarks out there are show their old performance, while they perform a lot better today.

With the latest optimizations, this combination is really fast. To show how fast, I've run some benchmarks against our Python images.

Reply:
Musl was released 9 years ago which is young for an c library in my interpretation. I don't know the full history of Alpine. It's more in container aspect.